aws-rds-cfn-common/src/main/java/software/amazon/rds/common/util/TranslatorUtils.java

```diff
@@ -0,0 +1,36 @@
+package software.amazon.rds.common.util;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class TranslatorUtils {
+
+    public static<M> M difference(final M previousModel, final M desiredModel, Supplier<M> instanceSupplier) {
```

Comment:
Is there any good library that can provide us same functionality?
I believe this could be popular issue. I would prefer to use library as reflection bugs are tough to discover and investigate. Also using good library could help here in future as this logic will be a good candidate to be used in post resource stabilization comparison  and logging resource differences as well.